broker/src/main/java/io/zeebe/broker/logstreams/LogStreamDeletionService.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.logstreams;
+
+import io.zeebe.broker.logstreams.state.StatePositionSupplier;
+import io.zeebe.logstreams.log.LogStream;
+import io.zeebe.logstreams.state.Snapshot;
+import io.zeebe.logstreams.state.SnapshotDeletionListener;
+import io.zeebe.logstreams.state.SnapshotStorage;
+import io.zeebe.servicecontainer.Injector;
+import io.zeebe.servicecontainer.Service;
+import io.zeebe.servicecontainer.ServiceStartContext;
+import io.zeebe.servicecontainer.ServiceStopContext;
+import io.zeebe.util.sched.Actor;
+import java.util.Objects;
+
+public class LogStreamDeletionService extends Actor
+    implements Service<SnapshotDeletionListener>, SnapshotDeletionListener {
+  private final Injector<SnapshotStorage> snapshotStorageInjector = new Injector<>();
+  private final LogStream logStream;
+  private final StatePositionSupplier positionSupplier;
+
+  private boolean isClosing;
+  private SnapshotStorage snapshotStorage;
+
+  public LogStreamDeletionService(
+      final LogStream logStream, final StatePositionSupplier positionSupplier) {
+    this.logStream = logStream;
+    this.positionSupplier = positionSupplier;
+  }
+
+  @Override
+  public void start(final ServiceStartContext startContext) {
+    snapshotStorage =
+        Objects.requireNonNull(
+            getSnapshotStorageInjector().getValue(), "must have a snapshot storage");
+    startContext.async(startContext.getScheduler().submitActor(this));
+  }
+
+  @Override
+  public void stop(final ServiceStopContext stopContext) {
+    stopContext.async(actor.close());
+  }
+
+  @Override
+  public SnapshotDeletionListener get() {
+    return this;
+  }
+
+  @Override
+  protected void onActorStarting() {
+    isClosing = false;
+    snapshotStorage.addDeletionListener(get());
+  }
+
+  @Override
+  protected void onActorClosing() {
+    isClosing = true;
+
+    if (snapshotStorage != null) {
+      snapshotStorage.removeDeletionListener(get());
+    }
+  }
+
+  @Override
+  public void onSnapshotDeleted(final Snapshot snapshot) {
+    actor.run(() -> delegateDeletion(snapshot));
```

Comment:
when the actor is in a closing phase jobs which are submitted are not executed anymore

Reply:
We discard those jobs? So I am guaranteed when close is requested that even if another thread calls `onSnapshotDeleted` it will not execute any job enqueued here?

Reply:
https://github.com/zeebe-io/zeebe/blob/develop/util/src/main/java/io/zeebe/util/sched/ActorTask.java#L282
And jobs are only executed in phase STARTED